### ISSUE TITLE
Refactor history item to remove top buttons

### ISSUE THREAD

**Issue by mcarare** [mozilla-mobile/fenix#17124]
Noticed on checking a11y issue https://github.com/mozilla-mobile/fenix/issues/16027.

Currently, a history item layout contains the `LibrarySiteItemView` but also the layouts for the top buttons/layouts ("Delete all" and "Recently closed tabs") that are shown only once on this screen. This could hurt performance and the layout could be improved by using a different `ViewHolder` for these elements or a `ConcatAdapter` (this might also break a11y behavior).

We should investigate what could be the best solution for this screen, in terms of performance, maintainability, and a11y.

**Comment by CharlesAdam** [comment]
is this still relevant?

**Comment by mcarare** [comment]
@CharlesAdam Sure, you can work on this.

**Comment by CharlesAdam** [comment]
@mcarare I was testing the a11y before any changes to the code, and I've noticed the history item "more options" list buttons doesn't announces Button.

I'm testing on Android 11 with TalkBack
![20210218_150726.gif](https://user-images.githubusercontent.com/43017375/108402115-e05f3d00-71fb-11eb-8ed8-8f304e56a084.gif)

**Comment by mcarare** [comment]
If you mean the list entries, those should be not announced as `Button`, and they are also not part of this issue.
If you mean to the actual "more options" button, for me it is announced as `Button`.

**Comment by CharlesAdam** [comment]
Ok, sorry for the confusion. 
It would be great to use ConcatAdapter, however, it is on the beta version of the recycler adapter lib, so I think it would be safer to use two ViewHolders

**Comment by neha-b2001** [comment]
Hello @eliserichards, could I work on this issue? Since I have made PRs for my previous issues...
